### PR TITLE
Fullscreen implementation

### PR DESCRIPTION
**Pull request by MeiHouwang**
with example

### CODE REVIEW

File: Sources/Display/Platform/Win32/win32_window.cpp [hunk 1]
```diff
@@ -1685,8 +1735,6 @@ void Win32Window::create_hid_devices()
 
 void Win32Window::get_styles_from_description(const DisplayWindowDescription &desc, DWORD &style, DWORD &ex_style)
 {
-	style = 0;
-	ex_style = 0;
```
Comment (dpjudas) [inline Sources/Display/Platform/Win32/win32_window.cpp]: Why is this needed? Without those two lines the flags might not be set correctly if a caller hasn't zero-initialized the out arguments.

Comment (MeiHouwang) [inline Sources/Display/Platform/Win32/win32_window.cpp]: Because caller could initialize arguments with his own styles already and this lines will override predefined styles. E.g. I always init window style with WS_OVERLAPPEDWINDOW|WS_VISIBLE, and these lines will override all predefined styles. I thought the purpose of this function was to add style bits, not to override style completely. Correct me if I'm wrong.

Comment (rombust) [inline Sources/Display/Platform/Win32/win32_window.cpp]: Yes, I see your point. It is actually confusing at the moment, due to the caller using 
		DWORD style = 0, ex_style = 0;
		get_styles_from_description(window_desc, style, ex_style);
I'll commit, and clarify the offending function

Comment (dpjudas) [inline Sources/Display/Platform/Win32/win32_window.cpp]: Actually I meant those arguments as purely out arguments rather than in/out. I really wish C++ had real qualifiers that could indicate this, or easily support multiple return arguments.

About the caller also initializing the fields, that is a coding practice I always use to ensure no variable is ever left in an uninitialized state. Basically both places function as a guard in case one of them failed to clear the variables before use.
